woocommerce/woocommerce
# CYS - Core: install font when user clicks opt-in

> [!WARNING]  
> This PR is blocked by #45586

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #45476 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure you have the WooCommerce Beta Tester plugin installed and activated
2. Under Tools > WCA Test Helper > Features, make sure `customize-store` is enabled
3. Under Tools > WCA Test Helper > Tools, click on `Reset Customize Your Store`-
5. Visit `wp-admin/admin.php?page=wc-settings&tab=advanced&section=woocommerce_com`
6. Ensure that the `Allow usage of WooCommerce to be tracked` is disabled.
7. Visit `wp-admin/admin.php?page=wc-admin&path=%2Fcustomize-store`.
8. Click on start `Start designing`.
9. On the sidebar, click on `Change homepage`.
10. Select a new homepage.
11. Save the change.
12. On the sidebar, click on `Choose Fonts`.
13. Check `Opt in to usage tracking to get access to more fonts`.
14. Click on `usage tracking`.
15. Click on Opt-in.
16. Ensure that the pre-load screen appears
17. Ensure that the Assembler is loaded
18. Click on the "Choose Fonts"
19. Ensure that multiple fonts are available.
20. Ensure that the homepage hasn't been restored.
21. Install https://wordpress.org/plugins/wp-data-access/
22. Visit wp-admin
23. On the sidebar WP Data Access > Query Builder
24. Copy paste: 
```
SELECT * FROM `wp_posts` WHERE `post_type` LIKE 'wp_font_family' ```

25. Click Execute
26. Ensure that a table with some results is visible
27. Copy paste:
```
SELECT * FROM `wp_posts` WHERE `post_type` LIKE 'wp_font_face' 
```
28. Ensure that a table with some results is visible
29. Remove them via:
```
DELETE FROM `wp_posts` WHERE `post_type` = 'wp_font_face';
DELETE FROM `wp_posts` WHERE `post_type` = 'wp_font_family';
```
30. Ensure that the `woocommerce_allow_tracking` option (`wp_option` table) is set to `yes`.
31. Update the option to `no`.
32. Refresh the assembler page.
33. Repeat 9-20.




### Changelog entry




<details>

#### Significance




#### Type




#### Message 
CYS - Core: install font when user clicks opt-in

#### Comment 


</details>
